File: minimal.hs
import Data.Map (fromListWith, (!))
import Data.Set (union, singleton, size)
import Data.Text (pack, unpack, split)
import System.Environment

type Pair = (Int, Int)

parseInt x = read x::Int

parse :: String -> Pair
parse = tuplify . map (parseInt . unpack) . split (=='|') . pack
        where tuplify [x,y] = (x,y)

unparse :: Pair -> String
unparse (x,y) = show x ++ "|" ++ show y

multipair :: Int -> [String] -> [String]
multipair k lines = let pairs    = map parse lines
                        tr (x,y) = (y, singleton x)
                        invmap   = fromListWith union $ map tr pairs
                        oc (_,y) = (>=k) $ size $ invmap ! y
                     in map unparse $ filter oc pairs
main = do
  args <- getArgs
  let n = parseInt $ head args 
   in interact $ unlines . multipair n . lines

File: minimal.js
input = prompt().split('\n');
map = {};
for (i in input) {
	input[i] = input[i].split('|');
	k = input[i][0];
	v = input[i][1];
	map[k] ? map[k].push(v) : map[k] = [v];
}
for (i in input)
	map[input[i][0]].length > 1 &&
	console.log(input[i][0]+'|'+input[i][1]);

